atomate/vasp/workflows/tests/test_exchange_workflow.py

```diff
@@ -0,0 +1,64 @@
+# coding: utf-8
+
+from __future__ import division, print_function, unicode_literals, absolute_import
+
+import os
+import unittest
+import pandas as pd
+
+from monty.os.path import which
+
+from atomate.vasp.workflows.base.exchange import ExchangeWF
+from atomate.vasp.firetasks.parse_outputs import (
+    MagneticDeformationToDb,
+    MagneticOrderingsToDb,
+)
+from atomate.utils.testing import AtomateTest, DB_DIR
+
+from json import load
+from pymatgen import Structure
+
+__author__ = "Nathan C. Frey"
+__email__ = "ncfrey@lbl.gov"
+
+module_dir = os.path.join(os.path.dirname(os.path.abspath(__file__)))
+db_dir = os.path.join(module_dir, "..", "..", "..", "common", "test_files")
+test_dir = os.path.join(module_dir, "..", "..", "test_files", "exchange_wf")
+
+
+class TestExchangeWF(AtomateTest):
+    @classmethod
+    def setUpClass(cls):
+        cls.Mn3Al = pd.read_json(os.path.join(test_dir, "Mn3Al.json"))
+        cls.db_file = ""
+        cls.uuid = 1
+        cls.structures = [Structure.from_dict(s) for s in cls.Mn3Al.structure]
+        cls.parent_structure = cls.structures[0]
+        cls.energies = [
+            e * len(cls.parent_structure) for e in cls.Mn3Al.energy_per_atom
+        ]
+        cls.cutoff = 3.0
+        cls.tol = 0.04
+        cls.heisenberg_settings = {"cutoff": 3.0, "tol": 0.04, 
+            "average": False}
+        cls.mc_settings = {"mc_box_size": 3, "equil_timesteps": 10, "mc_timesteps": 10}
+        cls.db_file = os.path.join(db_dir, "db.json")
+
+    def test_workflow(self):
+
+        c = {}
+        c["heisenberg_settings"] = self.heisenberg_settings
+        c["mc_settings"] = self.mc_settings
+        c["DB_FILE"] = self.db_file
+
+        wf = ExchangeWF(
+            magnetic_structures=self.structures,
+            energies=self.energies,
+            db_file=self.db_file,
+        ).get_wf(c=c)
+
+        self.assertEqual(wf.name, "Mn3Al - Exchange")
```

Comment:
Can you expand this into proper workflow testing. Currently the workflow is not run to check whether all fireworks/firetasks execute correctly. Once the workflow has finished you should examine the database at the end of the workflow to check that:

i) The workflow status is COMPLETED and not FIZZLED.
ii) The magnetism/exchange collections have the correct data in them.

See TestFerroelectricWorkflow for an example of how to do this (a lot of the complexity in that workflow is handling `RunFakeVasp` which you will not have in your workflow). Similar to your other tests, you can use `@requires` to avoid running the tests if vampire is not installed.

It would be good to get vampire installed on the testing environment. Is it available on conda?

Reply:
Unfortunately, I don't think it's available through conda. The build procedure is pretty simple though, and the C++ code is here: https://github.com/richard-evans/vampire